src/main/java/core/mate/academy/service/MachineProducerStrategy.java

```diff
@@ -0,0 +1,23 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Machine;
+import core.mate.academy.service.producerimpl.BulldozerProducerImpl;
+import core.mate.academy.service.producerimpl.ExcavatorProducerImpl;
+import core.mate.academy.service.producerimpl.TruckProducerImpl;
+import java.util.Optional;
+
+public class MachineProducerStrategy {
+    public Optional<MachineProducer<?>> getMachineProducerStrategy(Class<? extends Machine> type) {
+        switch (type.getSimpleName()) {
+            case "Bulldozer":
+                return Optional.of(new BulldozerProducerImpl());
+            case "Excavator":
+                return Optional.of(new ExcavatorProducerImpl());
+            case "Truck":
+                return Optional.of(new TruckProducerImpl());
+            default:
+                return Optional.empty();
+        }
```

Comment:
`Optional` is usually used when you know that something is nullable

in this case usage of `Optional` is unjustified - you may add some default implementation of `MachineProducer` for that purpose or return one from existing